### ISSUE TITLE
Remove workload auto-restart function when pvc expanded

### ISSUE THREAD

**Issue by stoneshi-yunify** [kubesphere/kubesphere#4090]
**Describe the Bug**
Need to remove the controller function in kubesphere/pkg/controller/storage/expansion

We can not stop a workload (by scaling down to replicas=0) without user's confirmation, because it may interrupt user's application and affect user's business.  And current function can only handle deployments/statefulsets, so it's incomplete. 

And there are some storage supporting online filesystem expansion that doesn't require pod restart.

**Versions Used**
KubeSphere: v3.1.1 and before
Kubernetes: (If KubeSphere installer used, you can skip this)

/kind bug
/area storage
/milestone v3.2
/good-first-issue

**Comment by wenhuwang** [comment]
@stoneshi-yunify  Does the removal of the workload automatic restart function require compatibility with some storage that does not support hot expansion, or for this situation to allow the user to manually restart.

**Comment by stoneshi-yunify** [comment]
> @stoneshi-yunify Does the removal of the workload automatic restart function require compatibility with some storage that does not support hot expansion, or for this situation to allow the user to manually restart.

Yes, user needs to decide when to restart the workload and restart it manually.

**Comment by wenhuwang** [comment]
> > @stoneshi-yunify Does the removal of the workload automatic restart function require compatibility with some storage that does not support hot expansion, or for this situation to allow the user to manually restart.
> 
> 
> 
> Yes, user needs to decide when to restart the workload and restart it manually. 

So we need to remove storage expansion controller？i have confirmed that the expansion controller is mainly to restart the workload.

**Comment by stoneshi-yunify** [comment]
> > > @stoneshi-yunify Does the removal of the workload automatic restart function require compatibility with some storage that does not support hot expansion, or for this situation to allow the user to manually restart.
> > 
> > 
> > Yes, user needs to decide when to restart the workload and restart it manually.
> 
> So we need to remove storage expansion controller？i have confirmed that the expansion controller is mainly to restart the workload.

Yes, simply remove the storage expansion controller will resolve this issue.
